docs/pages/command_bus.md

```diff
@@ -0,0 +1,332 @@
+
+    #[Handle]
+    public function changeName(ChangeProfileName $command): void
+    {
+        if (!$nameValidator($command->name) {
+            throw new InvalidArgument();
+        }
+
+        $this->recordThat(new NameChanged($command->name));
+    }
+
+    // ... apply methods
+}
+```
+
+#### Inject Service
+
+You can inject services into aggregate handler methods.
+Starting with the second parameter, it automatically tries to inject the service using a service locator. 
+Standard, it uses the fully qualified class name from the parameter type hint to find the service.
+
+```php
+use Patchlevel\EventSourcing\Aggregate\BasicAggregateRoot;
+use Patchlevel\EventSourcing\Attribute\Aggregate;
+use Patchlevel\EventSourcing\Attribute\Handle;
+use Psr\Clock\ClockInterface;
+
+#[Aggregate('profile')]
+final class Profile extends BasicAggregateRoot
+{
+    #[Handle]
+    public static function create(
+        CreateProfile $command,
+        ClockInterface $clock,
+    ): self {
+        $self = new self();
+
+        $self->recordThat(new ProfileCreated(
+            $command->id,
+            $command->name,
+            $clock->now(),
+        ));
+
+        return $self;
+    }
+
+    // ... apply methods
+}
+```
+
+!!! note
+
+    The service must be registered in the service locator.
+
+!!! tip
+
+    You can inject multiple services into the handler method.
+
+Or you can inject the service manually using the `#[Inject]` attribute.
+There you can specify the service name that should be injected.
+
+```php
+use Patchlevel\EventSourcing\Aggregate\BasicAggregateRoot;
+use Patchlevel\EventSourcing\Attribute\Aggregate;
+use Patchlevel\EventSourcing\Attribute\Handle;
+use Patchlevel\EventSourcing\Attribute\Inject;
+
+#[Aggregate('profile')]
+final class Profile extends BasicAggregateRoot
+{
+    #[Handle]
+    public static function create(
+        CreateProfile $command,
+        #[Inject('name_validator')]
+        NameValidator $nameValidator,
+    ): self {
+        $self = new self();
+
+        if (!$nameValidator($command->name) {
+            throw new InvalidArgument();
+        }
+
+        $self->recordThat(new ProfileCreated($command->id, $command->name));
+
+        return $self;
+    }
+
+    // ... apply methods
+}
+```
+
+!!! note
+
+    Injection in handler methods is only possible with the `AggregateHandlerProvider`.
+
+## Setup
+
+We provide a `DefaultCommandBus` that you can use to dispatch commands.
+You need to pass a `HandlerProvider` to the constructor.
+
+```php
+use Patchlevel\EventSourcing\CommandBus\DefaultCommandBus;
+use Patchlevel\EventSourcing\CommandBus\HandlerProvider;
+
+/**
+ * @var HandlerProvider $handlerProvider
+ */
+$commandBus = new DefaultCommandBus($handlerProvider);
+
+$commandBus->dispatch(new CreateProfile($profileId, 'name'));
+$commandBus->dispatch(new ChangeProfileName($profileId, 'new name'));
+```
+
+!!! note
+
+    The `DefaultCommandBus` is a synchronous command bus. 
```

Comment:
Maybe we could name it `SyncCommandBus` then?

Reply:
I don't know, I don't really like the name.

Reply:
I think I would stick with that name. We also have a `DefaultEventBus`.

Reply:
Yes i know, but also there the naming is not good IMHO. We also have an open issue about the naming problem https://github.com/patchlevel/event-sourcing/issues/230.

`SyncCommandBus` is better describing what the class is actually doing. You even noted it here in the docs so that the user is aware of it. If it would be named like this, the note wouldnt be necessary as the user already knows it.

This is not a hard blocker from my side, but i think we should not add more `Default*` classes if we can name it better.

Reply:
The note would stay the same, as the point is to emphasize that a command is always fully processed before another command is executed.

But if you want, I can rename it.